enhancements/monitoring/multi-tenant-alerting.md

```diff
@@ -0,0 +1,601 @@
+(`alertmanagerConfigSelector` and `alertmanagerConfigNamespaceSelector`) to
+select which `AlertmanagerConfig` resources should be reconciled by the
+Prometheus operator and from which namespace(s). Before this proposal, the
+Plaform Alertmanager resource defines the 2 selectors as null, meaning that it
+doesn't select any `AlertmanagerConfig` resources.
+
+We propose to add a new boolean field `enableUserAlertmanagerConfig` to the
+`openshift-montoring/cluster-monitoring-config` configmap. If
+`enableUserAlertmanagerConfig` is missing, the default value is false.
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |-
+    enableUserWorkload: true
+    alertmanager:
+      enableUserAlertmanagerConfig: true
+```
+
+When `enableUserAlertmanagerConfig` is true, the cluster monitoring operator
+configures the Platform Alertmanager to reconcile `AlertmanagerConfig`
+resources from user namespaces as follows.
+
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: Alertmanager
+metadata:
+  name: main
+  namespace: openshift-monitoring
+spec:
+  alertmanagerConfigSelector: {}
+  alertmanagerConfigNamespaceSelector:
+    matchExpressions:
+    - key: openshift.io/cluster-monitoring
+      operator: NotIn
+      values:
+      - "true"
+    - key: openshift.io/user-monitoring
+      operator: NotIn
+      values:
+      - "false"
+  ...
+```
+
+To be consistent with what exists already for service/pod monitors and rules,
+the Prometheus operator doesn't reconcile `AlertmanagerConfig` resources from
+namespaces with the `openshift.io/user-monitoring: "false"` label. It allows
+application owners to opt out completely from UWM in case they deploy and run
+their own monitoring infrastructure (for instance with the [Monitoring Stack
+operator][monitoring-stack-operator]).
+
+In addition, the cluster admins can exclude specific user namespace(s) from UWM
+with the new `excludeUserNamespaces` field.
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |-
+    enableUserWorkload: true
+    alertmanager:
+      enableUserAlertmanagerConfig: true
+    excludeUserNamespaces: [foo,bar]
+```
+
+##### Dedicated UWM Alertmanager
+
+In some environments where cluster admins and UWM admins are different personas
+(e.g. OSD), it might not be acceptable for cluster admins to let users
+configure the Platform Alertmanager with `AlertmanagerConfig` resources because:
+* User configurations may break the Alertmanager configuration.
+* Processing of user alerts may slow down the alert notification pipeline.
+* Cluster admins don't want to deal with delivery errors for user notifications.
+
+At the same time, application owners want to configure their alert
+notifications without requesting external intervention.
+
+In this case, UWM admins have the possibility to deploy a dedicated
```

Comment:
This conflicts with https://github.com/openshift/enhancements/pull/937/files#diff-02ac51e89242574f67f8eb856975563d598d4b8bbdc18f7ce848ed135a9a8de5R87 ?

Reply:
correct I will remove it from the non-goals.